File: backend/internal/api/tuner_status_test.go
package api

import (
	"testing"

	"github.com/fuba/tv-viewer/internal/mirakurun"
)

func TestSummarizeTunersSeparatesViewerEPGAndOtherUsage(t *testing.T) {
	tuners := []mirakurun.Tuner{
		{
			Types: []string{"GR"}, IsUsing: true,
			Users: []mirakurun.TunerUser{
				{ID: "Mirakurun:getEPG()"},
				{Agent: "tv-viewer/1.0", URL: "/api/channels/GR/16/stream"},
			},
		},
		{
			Types: []string{"GR"}, IsUsing: true,
			Users: []mirakurun.TunerUser{{Agent: "BBA/0.1.0"}},
		},
		{
			Types: []string{"GR"}, IsFree: true,
			Users: []mirakurun.TunerUser{{Agent: "tv-viewer/1.0"}},
		},
		{
			Types: []string{"BS", "CS"}, IsUsing: true,
			Users: []mirakurun.TunerUser{{ID: "Mirakurun:getEPG()"}},
		},
	}

	got := summarizeTuners(tuners)
	if len(got) != 2 {
		t.Fatalf("status count = %d, want 2", len(got))
	}
	if got[0] != (TunerStatus{Type: "GR", Total: 3, Using: 2, Free: 1, ViewerUsing: 1, EPGUsing: 1, OtherUsing: 1, ViewerShared: 1}) {
		t.Fatalf("GR status = %+v", got[0])
	}
	if got[1] != (TunerStatus{Type: "BS/CS", Total: 1, Using: 1, EPGUsing: 1}) {
		t.Fatalf("BS/CS status = %+v", got[1])
	}
}

File: frontend/src/lib/tunerStatus.test.ts
import assert from 'node:assert/strict'
import test from 'node:test'

import { tunerDetails, type TunerInfo } from './tunerStatus.ts'

test('tunerDetails explains viewer usage and physical sharing', () => {
  const tuner: TunerInfo = {
    type: 'GR', total: 4, using: 4, free: 0,
    viewerUsing: 1, epgUsing: 3, otherUsing: 1, viewerShared: 1,
  }
  assert.equal(
    tunerDetails(tuner),
    'TV Viewer 1台（うち共有1台） / EPG取得 3台 / その他 1台 / 空き 0台',
  )
})
